release: procedure change (#6148)
cosmos/cosmos-sdk: CONTRIBUTING.md

releases will be based off of that release.

- start on `vX.XX.X`

> - checkout a new branch `rc/vX.X.X`

Reviewer: ```suggestion
- checkout a new branch `rc0/vX.X.X`
```

- cherry pick the desired changes from `master`
  - these changes should be small and NON-BREAKING (both API and state machine)
- add entries to CHANGELOG.md and remove corresponding pending log entries

> - checkout a new branch `release/vX.X.X` based off of `vX.XX.X`

Reviewer: ```suggestion
- checkout a new branch `release/vX.X.X` based off of the previous release
```

> - create a PR merging `rc/vX.X.X` into `release/vX.X.X`

Reviewer: ```suggestion
- create a PR merging `rcN/vX.X.X` into `release/vX.X.X`
```

- create a PR merging `pre-rc/vX.X.X` into `rc/vX.X.X`
- checkout a new branch `release/vX.X.X` based off of `vX.XX.X`
- create a PR merging `rc/vX.X.X` into `release/vX.X.X`
- run tests and simulations (noted in [Release Procedure](#release-procedure))
- after tests and simulation have successfully completed, create the release branch `release/vX.XX.X` from the `RC` branch

> - delete the `pre-rc/vX.X.X` and `RC` branches

Reviewer: We still need to delete all the RC branches.

Author: i believe all branches are deleted after merging unless there is another branch depending on it or its protected

Reviewer: RC branches have to be manually deleted AFAIK. I could be wrong.
